libraries/ms-drivers/src/pcf8523_rtc.c

```diff
@@ -0,0 +1,79 @@
+#include "pcf8523_rtc.h"
+#include "bcd.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "i2c_mcu.h"
+#include "pcf8523_rtc_defs.h"
+#include "status.h"
+
+#define DEFAULT_CR1_SETTINGS (1 << TIME_12_24 | 1 << CAP_SEL)
+#define STOP_CR1_SETTINGS (1 << TIME_12_24 | 1 << STOP | 1 << CAP_SEL)
+
+static I2CPort port;
+
+StatusCode pcf8523_init(I2CPort i2c_port) {
+  port = i2c_port;
+  // Write the address of the first control register and then
+  // send the data for the 3 control registers
+  // Note that the addr of the control register auto-increments
+  // After each byte is sent
+  uint8_t data[NUM_CONTROL_REG + 1];
+  data[0] = CR1;
+  // Set 24 hr time and 12.5pF load capacitance
+  data[1] = DEFAULT_CR1_SETTINGS;
+  // Default settings for CR2 and CR3
+  data[2] = 0;
+  data[3] = 0;
+
+  return i2c_write(port, I2C_ADDR, data, (sizeof(data)));
+}
+
+StatusCode pcf8523_get_time(Pcf8523Time *time) {
+  // Set starting register address (this will auto-increment)
+  uint8_t starting_reg = SECONDS;
+  i2c_write(port, I2C_ADDR, &starting_reg, 1);
+
+  // Read time registers
+  uint8_t data[NUM_TIME_REG];
+  i2c_read(port, I2C_ADDR, data, (sizeof(data)));
+
+  // Store time
+  time->seconds = bcd_to_dec(data[0]);

```

Comment:
Bit 7 of the seconds register is `OS` for "oscillator stop" - it's 1 if "clock integrity is not guaranteed; oscillator has stopped or been interrupted". To avoid this condition showing up as weird numbers of seconds, you might want to handle the OS bit separately with a `LOG_WARN` or something, then set it to 0 for the seconds bcd conversion.